src/oscar/apps/partner/abstract_models.py

```diff
@@ -180,6 +180,11 @@ def net_stock_level(self):
             return self.num_in_stock
         return self.num_in_stock - self.num_allocated
 
+    @property
```

Comment:
I think [`cached_property`](https://docs.djangoproject.com/en/2.0/ref/utils/#django.utils.functional.cached_property) would be good here to ensure we don't query this join repeatedly?